cmd/operator-sdk/generate/bundle/bundle.go

```diff
@@ -0,0 +1,217 @@
+	}
+	if err := csvGen.Generate(cfg, opts...); err != nil {
+		return fmt.Errorf("error generating ClusterServiceVersion: %v", err)
+	}
+
+	if err := kustomize.WriteIfNotExist(c.outputDir, kubebuilderKustomization); err != nil {
+		return err
+	}
+
+	if !c.quiet {
+		fmt.Println("Bases generated successfully")
+	}
+
+	return nil
+}
+
+func (c bundleCmd) validateManifests(*config.Config) (err error) {
+	if c.version != "" {
+		if err := genutil.ValidateVersion(c.version); err != nil {
+			return err
+		}
+	}
+
+	if !genutil.IsPipeReader() {
+		if c.manifestRoot == "" {
+			return errors.New("--manifest-root must be set if not reading from stdin")
+		}
+		if c.crdsDir == "" {
+			return errors.New("--crds-dir must be set if not reading from stdin")
+		}
+	}
+
+	if c.stdout {
+		if c.outputDir != "" {
+			return errors.New("--output-dir cannot be set if writing to stdout")
+		}
+	}
+
+	return nil
+}
+
+func (c bundleCmd) runManifests(cfg *config.Config) (err error) {
+
+	if !c.quiet && !c.stdout {
+		if c.version == "" {
+			fmt.Println("Generating bundle manifests")
+		} else {
+			fmt.Println("Generating bundle manifests version", c.version)
+		}
+	}
+
+	defaultBundleDir := filepath.Join("config", "bundle")
+	if c.inputDir == "" {
+		c.inputDir = defaultBundleDir
+	}
+	if !c.stdout {
+		if c.outputDir == "" {
+			c.outputDir = defaultBundleDir
+		}
+	}
+	// Only regenerate API definitions once.
+	if c.apisDir == "" && !c.kustomize {
+		if cfg.MultiGroup {
+			c.apisDir = "apis"
+		} else {
+			c.apisDir = "api"
+		}
+	}
+
+	col := &collector.Manifests{}
+	if genutil.IsPipeReader() {
+		if err := col.UpdateFromReader(os.Stdin); err != nil {
+			return err
+		}
+	}
+	if c.manifestRoot != "" {
+		if err := col.UpdateFromDirs(c.manifestRoot, c.crdsDir); err != nil {
+			return err
+		}
+	}
+
+	csvGen := gencsv.Generator{
+		OperatorName: c.operatorName,
+		OperatorType: genutil.PluginKeyToOperatorType(cfg.Layout),
+		Version:      c.version,
+		Collector:    col,
+	}
+
+	// FIXME(estroz): this may not "upgrade" a bundled CSV because the kustomize
+	// base may not contain previous version information. The base is assumed to be "blank".
+	stdout := genutil.NewMultiManifestWriter(os.Stdout)
+	opts := []gencsv.Option{
+		gencsv.WithBase(c.inputDir, c.apisDir),
+	}
+	if c.stdout {
+		opts = append(opts, gencsv.WithWriter(stdout))
+	} else {
+		opts = append(opts, gencsv.WithBundleWriter(c.outputDir))
+	}
+
+	if err := csvGen.Generate(cfg, opts...); err != nil {
+		return fmt.Errorf("error generating ClusterServiceVersion: %v", err)
+	}
+
+	if c.stdout {
+		if err := genutil.WriteCRDs(stdout, col.CustomResourceDefinitions...); err != nil {
+			return err
+		}
+	} else {
+		dir := filepath.Join(c.outputDir, bundle.ManifestsDir)
+		if err := genutil.WriteCRDFiles(dir, col.CustomResourceDefinitions...); err != nil {
+			return err
+		}
+	}
+
+	if !c.quiet && !c.stdout {
+		fmt.Println("Bundle manifests generated successfully")
```

Comment:
it is very helpful to know what and where was the file generated. Could we improve the log info?

Reply:
@estroz and @camilamacedo86 We should consolidate further review comments on #3065 so we don't end up suggesting the same changes in both places.

@camilamacedo86 This is from your earlier suggestion to break this PR into smaller PRs.